src/translate.py

```diff
@@ -1007,42 +1004,71 @@ class StructAccess(Expression):
     struct_var: Expression
     offset: int
     target_size: Optional[int]
-    field_name: Optional[str] = field(compare=False)
+    field_path: Optional[AccessPath] = field(compare=False)
     stack_info: StackInfo = field(compare=False, repr=False)
     type: Type = field(compare=False)
-    has_late_field_name: bool = field(default=False, compare=False)
+    checked_late_field_path: bool = field(default=False, compare=False)
+
+    @staticmethod
+    def access_path_to_field_name(path: Optional[AccessPath], deref: bool) -> str:
+        """
+        Convert a path like `["foo", 3, "bar"]` into a field name like `.foo[3].bar`.
+        If deref is True, the initial `.` is replaced by `->`.
```

Comment:
if deref is true and path is empty or has an integer as first component, this fails. Should that be asserted against or signalled via return value?

Reply:
Added an assert, clarified the docstring, and made `StructAccess.format()` super explicit to avoid this case. 
(This case wasn't supposed to happen before, but the checks I had in format weren't specific enough)